src/components/Button/Button.jsx

```diff
@@ -2,23 +2,40 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './Button.module.css';
 
-const Button = (props) => (
-  <button
-    className={styles.main}
-    onClick={props.onClick}
-    type="button"
-  >
-    {props.children}
-  </button>
-);
+export default function Button(props) {
+  const {
+    btnRole, text, onClick, disabled,
+  } = props;
+  if (btnRole === 'submit') {
```

Comment:
why it did like this?

Reply:
At first my intention was to dynamically change button's type attribute, but it turned out that it seems to be an anti-pattern and better to use if/else statement and hardcode several buttons.
https://github.com/yannickcr/eslint-plugin-react/issues/1555#issuecomment-377366789